### DISCUSSION TITLE
Browser feature idea: expand on related streams

### DISCUSSION THREAD

**Discussion by piotrekbator** [RailsEventStore/rails_event_store#1210] [Ideas]
Recently, when working on linking events to some wider, more general streams I wanted to use the related streams feature of the RES Browser and was quite surprised by how it worked.

I was absolutely sure that the related streams would appear in the event details view and only after some digging I found out that they only appear in the stream view. That's fine and still quite helpful, but I see room for improvement, which I'd be happy to contribute as well.

What I'm proposing is:

1. Let's start displaying all the streams an event is in in the event details view, in the "related streams" section, where the generic correlation and by type streams appear.
2. Let's also expand this list by running each of the streams by the related streams finder if it's defined.

My motivation for this is following:

When I land on a specific event in the browser I'd like to be able to go straight to the stream this event is in to gain some context. Now the only way to do this is to know how the streams are named, construct the name manually and type it into the "go to stream" box. Which is just inconvenient for me and disqualifying for less-technical users that also have access to the browser in my case.

It's possible that there's a reason why it's currently implemented like it is. I can imagine that looking up event's streams whenever we look at an event in the Browser might cause performance issues. Or there might be other reasons. If that's the case, just let me know.

But if you think this feature would be beneficial for others as well and don't see any obvious drawbacks, I can give it a shot and try implementing this in a PR. 

Let me know what you think.

**Comment by mostlyobvious** [comment]
That's a good idea and proposal. We want to make browser much powerful tool for everyone and it's a step in that direction.

The performance shouldn't be an issue here. Even if that's not the fastest call/query, the benefits for debugging and understanding the system are much greater.

On `Client` there's already an API for discovering event streams — `streams_of(event_id)`. It would need exposure in browser API.

Feel free to explore this topic and propse PR!

Related: https://github.com/RailsEventStore/rails_event_store/issues/626#issuecomment-750105125

**Comment by piotrekbator** [comment]
Perfect. I'll give it a try and come back with a PR when something comes out of it.

I'll need to have a look, but probably after this change the default related streams could be removed. They'd work if someone had them configured as mentioned in #626 and would disappear when not used.
